src/status_im/ui/screens/browser/navigation.cljs

```diff
@@ -1,6 +1,8 @@
 (ns status-im.ui.screens.browser.navigation
-  (:require [status-im.ui.screens.navigation :as navigation]))
+  (:require [status-im.ui.screens.navigation :as navigation]
+            [status-im.models.browser-history :as browser-history]))
 
 (defmethod navigation/preload-data! :browser
   [db [_ _ {:keys [browser/browser-id]}]]
-  (assoc db :browser/options {:browser-id browser-id}))
+  (let [dont-store (browser-history/dont-store-history-on-nav-change-if-history-exists db browser-id)]
+    (assoc db :browser/options (merge {:browser-id browser-id} dont-store))))
```

Comment:
Do you actually need a `merge` here? (if don't store might have a different value for `browse-id`), otherwise assoc will do, swapping the two:
`(assoc dont-store :browser-id browser-id)`

Reply:
Sure, will use `assoc` :+1:

Reply:
Done!